pkg/scheduler/backend/queue/active_queue.go

```diff
@@ -201,7 +217,8 @@ func (aq *activeQueue) pop(logger klog.Logger) (*framework.QueuedPodInfo, error)
 	aq.schedCycle++
 	// In flight, no concurrent events yet.
 	if aq.isSchedulingQueueHintEnabled {
-		aq.inFlightPods[pInfo.Pod.UID] = aq.inFlightEvents.PushBack(pInfo.Pod)
```

Comment:
So, the problem of the previous implementation is here.
If `aq.inFlightPods[pInfo.Pod.UID]` already has the list.Element, here a new Element will overwrite it and the old one won't be cleaned up forever from `inFlightEvents`.

Reply:
if the Pod is already inflight, maybe it would be better just to discard it at this point (and report the issue in the log)

Reply:
If we discard the duplicated Pod here, the next point would be whether we should requeue the pod to activeQ/backoffQ when in-flight pod is back to the queue... Probably, we should, to be on the safe side, then how.

Reply:
> Probably, we should, to be on the safe side, but then how.

Ah, okay, I noticed, when we really should, in-flight events recorded for the pod should requeue the pod to activeQ/backoffQ. Okay, then I think it's safe to just discard the duplicated pod at Pop(). I'll update the PR.